misk-events/src/main/kotlin/misk/events/Producer.kt

```diff
@@ -0,0 +1,25 @@
+package misk.events
+
+/**
+ * A [Producer] is used to send events to an event stream.
+ */
+interface Producer {
+  /**
+   * A producer [Transaction] is a unit of publishing. Producer transactions adhere to the
+   * transactional concepts of isolation (events publishing within a transaction are only
+   * visible to consumers once the transaction is committed) and atomicity (once committed, all
+   * events within the transactions are made available to consumers - no events will be lost).
+   *
+   * Note that a Producer transaction is _not_ connected to a database transaction; it is solely
+   * a transaction within the event streaming system. To coordinate event publishing with
+   * local database transactions, use a [SpooledProducer] which stores the events in a local table
+   * as part of the local database transaction
```

Comment:
worthwhile to document behavior of long-lived transactions? Do transactions ever time out/auto-rollback?

Reply:
Sure